stub/src/main/java/io/grpc/stub/StreamObserver.java

```diff
@@ -54,9 +66,9 @@
   /**
    * Receives a terminating error from the stream.
    *
-   * <p>May only be called once and if called it must be the last method called. In particular if an
-   * exception is thrown by an implementation of {@code onError} no further calls to any method are
-   * allowed.
+   * <p>May only be called once and if called it must be the last method called. Although
+   * implementations are expected to not throw exceptions, if an exception is thrown by {@code
+   * onError()} further calls to the observer should be avoided.
```

Comment:
Saying "further calls should be avoided" sounds a bit odd/redundant to me coming immediately after "must be the last method called" (same for `onCompleted()`)